plugins/catalog/src/components/EntityPeekCard/EntityPeekCard.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useCallback } from 'react';
+import {
+  CompoundEntityRef,
+  Entity,
+  isUserEntity,
+  isGroupEntity,
+  UserEntity,
+  GroupEntity,
+} from '@backstage/catalog-model';
+import {
+  Button,
+  MenuItem,
+  Paper,
+  ListItemIcon,
+  ListItemText,
+  Divider,
+  makeStyles,
+  Menu,
+  Link,
+} from '@material-ui/core';
+import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
+import EmailIcon from '@material-ui/icons/Email';
+import { bindTrigger, bindMenu, InjectedProps } from 'material-ui-popup-state';
+import { catalogApiRef } from '@backstage/plugin-catalog-react';
+import { useApi } from '@backstage/core-plugin-api';
+import useAsync from 'react-use/lib/useAsync';
+import { usePopupState } from 'material-ui-popup-state/hooks';
+import { useNavigate } from 'react-router';
+import { Progress } from '@backstage/core-components';
+
+export type EntityPeekCardProps = {
+  entityRefs: CompoundEntityRef[];
```

Comment:
The fact that the entity could have multiple owners complicates this drop down a little. I suppose it's mostly going to only have one entity? What I have tried to do is iterate over the entities and provide menu items for each.